napari/utils/transforms/transforms.py

```diff
@@ -503,3 +499,185 @@ def expand_dims(self, axes: Sequence[int]) -> 'Affine':
+        return _apply_affine_with_padding(
+            coords, self._linear_matrix, self._translate
+        )
+
+    def __array__(self, *args, **kwargs):
+        """NumPy __array__ protocol to get the affine transform matrix."""
+        return self.affine_matrix
+
+    @property
+    def ndim(self) -> int:
+        """Dimensionality of the transform."""
+        return self._linear_matrix.shape[0]
+
+    @property
+    def translate(self) -> np.array:
+        """Return the translation of the transform."""
+        return self._translate
+
+    @translate.setter
+    def translate(self, translate):
+        """Set the translation of the transform."""
+        self._translate = coerce_translate(translate, self.ndim)
+        self._linear_matrix = self._make_linear_matrix()
+
+    @property
+    def scale(self) -> np.array:
+        """Return the scale of the transform."""
+        return self._scale
+
+    @scale.setter
+    def scale(self, scale):
+        """Set the scale of the transform."""
+        self._scale = coerce_scale(scale, self.ndim)
+        self._linear_matrix = self._make_linear_matrix()
+
+    @property
+    def rotate(self) -> np.array:
+        """Return the rotation of the transform."""
+        return self._rotate
+
+    @rotate.setter
+    def rotate(self, rotate):
+        """Set the rotation of the transform."""
+        self._rotate = coerce_rotate(rotate, self.ndim)
+        self._linear_matrix = self._make_linear_matrix()
+
+    @property
+    def shear(self) -> np.array:
+        """Return the shear of the transform."""
+        if is_matrix_upper_triangular(self._shear):
+            return self._shear[np.triu_indices(n=self.ndim, k=1)]
+        return self._shear
+
+    @shear.setter
+    def shear(self, shear):
+        """Set the shear of the transform."""
+        self._shear = coerce_shear(shear, self.ndim)
+        self._linear_matrix = self._make_linear_matrix()
+
+    @property
+    def affine_matrix(self) -> np.array:
+        """Return the affine matrix for the transform."""
+        matrix = np.eye(self.ndim + 1, self.ndim + 1)
+        matrix[:-1, :-1] = self._linear_matrix
+        matrix[:-1, -1] = self._translate
+        return matrix
+
+    @affine_matrix.setter
+    def affine_matrix(self, affine_matrix):
+        """Set the affine matrix for the transform."""
+        self._linear_matrix = affine_matrix[:-1, :-1]
+        self._translate = affine_matrix[:-1, -1]
+
+    @property
+    def inverse(self) -> 'Affine':
+        return Affine(affine_matrix=np.linalg.inv(self.affine_matrix))
+
+    def compose(self, transform) -> 'Affine':
+        """Return the composition of this transform and the provided one.
+
+        The provided transform must have an affine_matrix property/attribute.
+        """
+        affine_matrix = self.affine_matrix @ transform.affine_matrix
+        return Affine(affine_matrix=affine_matrix)
+
+    def set_slice(self, axes: Sequence[int]) -> 'CompositeAffine':
```

Comment:
I don't know whether I'm opening a can of worms that was closed long ago, but `set_slice()` is a misleading name here, since nothing is actually set (I would assume, in-place). `def slice()` would make more sense here, imho.

Reply:
Agreed. Unsure if I should make the change in this PR (including deprecating `set_slice`), so let me know if I should.

Reply:
I don't have an opinion — as you prefer!

Reply:
Let's leave this for another PR. This one is big enough already.

Reply:
fine with me to come in a new PR